### PR TITLE
Sort modules and their attributes as desired in 'cfbs pretty'

### PR DESCRIPTION
**Pull request by vpodzime**
https://github.com/cfengine/build-index/blob/master/CONTRIBUTING.md#publishing-your-module-for-others-to-reuse
defines ordering for the modules attributes so `cfbs pretty`
should make sure they are in order.

Ticket: CFE-3834
Changelog: 'cfbs pretty' now sorts modules and their attributes

### CODE REVIEW

**Comment by vpodzime** [comment]
*black* suggests this change:
```
diff --git a/cfbs/main.py b/cfbs/main.py
index 141b37a..338d8a3 100644
--- a/cfbs/main.py
+++ b/cfbs/main.py
@@ -57,7 +57,8 @@ def get_args():
     )
     parser.add_argument(
         "--keep-order",
-        help="Keep order of items in the JSON in '%s pretty'" % os.path.basename(sys.argv[0]),
+        help="Keep order of items in the JSON in '%s pretty'"
+        % os.path.basename(sys.argv[0]),
         action="store_true",
     )
```
which I really don't like and I don't feel obeying to a formatter script.

**Review by larsewi — COMMENTED** [review]
It looks beautiful! But for mortals like my self, it's a bit hard to comprehend advanced wizardry like this. Could you please make a pytest proving that this works?

File: cfbs/main.py [hunk 1]
```diff
@@ -53,6 +55,11 @@ def get_args():
         default=None,
         help="Expected checksum of the downloaded file",
     )
+    parser.add_argument(
+        "--keep-order",
+        help="Keep order of items in the JSON in '%s pretty'" % os.path.basename(sys.argv[0]),
```
Comment (olehermanse) [inline cfbs/main.py:60]: How important is it to change the name of `cfbs` and have it show up differently in the help output?

IMO this is much more readable and greppable: 

```suggestion
        help="Keep order of items in the JSON in 'cfbs pretty'",
```

**Review by larsewi — APPROVED** [review]
Awesome!

**Review by olehermanse — APPROVED** [review]
Looks good, there was one small "inconsistency" inside the comment JSON, but no big deal.

File: cfbs/pretty.py [hunk 2]
```diff
@@ -1,36 +1,140 @@
...
+    ```
+    {
+      "something": ...,
+      "modules": {
+        "mod1": {
+          "attr1": ...,
+        },
+        ...
+      },
+      "something else",
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply cfbs/pretty.py:66]: I was actually trying to show that there can be other things than dicts/objects.

**Pull request merged**